ocaml/xapi/xapi_xenops.ml

```diff
@@ -647,29 +647,30 @@ module MD = struct
     let pci_address = Db.PCI.get_pci_id ~__context ~self:pci in
     Xenops_interface.Pci.address_of_string pci_address
 
+  let get_virtual_pci_address ~__context vgpu =
+    let open Pci in
+    let device = vgpu.Db_actions.vGPU_device |> int_of_string in
```

Comment:
Why do we need a conversion string -> int -> string -> int? At a minimum, `device` should be a string that is converted once to an integer. A string that is a hexadecimal number can be converted by `int_of_string( "0x"^str)` to an int.

Reply:
Since the device is string but decimal.
Then I have to at first convert it to a hex by Printf.sprintf. and then convert the hex string to int.
You are right that if the device is hex then the `int_of_string( "0x"^str)` is good.

Reply:
Thanks. 
But how about the device is string but decimal, it's not hex.
Then I had to convert it to hex by Printf.sprintf and then convert to int.

Reply:
What is the format of vgpu.Db_actions.vGPU_device? In either case, one conversion from string to int should be enough.

Reply:
vgpu.Db_actions.vGPU_device is string of decimal.
like device = 10, then the pci_address, should be 0000:0:0a.0.
So device should be string"10" ->10(int) ->"0a" (string) -> 0a(int). because of the type address is defined:
{
  domain: int
  bus:int
  dev:int
  fn:int
}

Reply:
The address record defines `dev` to be int, so a single `int_of_string` is enough. There is no such thing as `0a(int)`. Decimal and hexadecimal are string representations of an `int` value and you can convert the integer `dev` to whatever you need when you need a string.

Reply:
Thanks, after checking it's just working with int_of_string without the prefix.